File: PodrziMe.Model/Kategorija.cs
using System;
using System.Collections.Generic;

namespace PodrziMe.Model { 
    public class Kategorija
    {
        public int KategorijaId { get; set; }

        public string NazivKategorije { get; set; } = null!;

        public int? PodKategorijaId { get; set; }

        public virtual PodKategorija? PodKategorija { get; set; }
    }
}

File: UI/podrzime_mobile/lib/modals/kategorija.dart
import 'package:json_annotation/json_annotation.dart';

part 'kategorija.g.dart';

@JsonSerializable()
class Kategorija {
  int? kategorijaId;
  String? nazivKategorije;
  int? podKategorijaId;

  Kategorija({this.kategorijaId, this.nazivKategorije, this.podKategorijaId});

  factory Kategorija.fromJson(Map<String, dynamic> json) =>
      _$KategorijaFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$KategorijaToJson(this);
}
